base/core.bal

```diff
@@ -0,0 +1,471 @@
+#
+# + 'type - Type of search parameter  
+# + name - Name of the search parameter (Key of the query parameter)
+# + value - Original incoming search parameter value in string format
+# + typedValue - Parsed/Decoded search parameter value based on the type of the search parameter
+public type RequestSearchParameter record {
+    readonly string name;
+    readonly string value;
+    readonly FHIRSearchParameterType 'type;
+    readonly FHIRTypedSearchParameter typedValue;
+};
+
+# Base type of decoded/processed search parameter value.
+# 
+# + name - Name of the search parameter
+# + modifier - FHIR Search parameter modifier
+public type FHIRTypedSearchParameter record {
+    readonly string name;
+    readonly FHIRSearchParameterModifier|string? modifier;
+};
+
+# Number type search parameter information.
+#
+# + value - Value of the received search parameter
+# + prefix - prefix is used to control the nature of the matching
+public type NumberSearchParameter record {
+    *FHIRTypedSearchParameter;
+    readonly Prefix prefix = eq;
+    readonly int|float value;
+};
+
+# String type search parameter information.
+#
+# + value - Value of the received search parameter
+public type StringSearchParameter record {
+    *FHIRTypedSearchParameter;
+    readonly string value;
+};
+
+# URI type search parameter information.
+#
+# + uri - URI/URL/URN value
+public type URISearchParameter record {
+    *FHIRTypedSearchParameter;
+    readonly uri uri;
+};
+
+# Date type search parameter information.
+#
+# + prefix - Prefix attached with date type search parameter
+# + value - Value of the received search parameter
+public type DateSearchParameter record {
+    *FHIRTypedSearchParameter;
+    readonly Prefix prefix;
+    //readonly date|dateTime|instant|Period value;
+    readonly time:Civil value;
+};
+
+# Token type search parameter information.
+#
+# + system - The system property of the Identifier or Coding
+# + code - Value of Coding.code or Identifier.value
+public type TokenSearchParameter record {
+    *FHIRTypedSearchParameter;
+    readonly string? system;
+    readonly string? code;
+};
+
+# Reference type search parameter information.
+# 
+# + resourceType - Referenced resource type
+# + id - The logical [id] of a resource using a local reference (i.e. a relative reference)
+# + url - Absolute URL - a reference to a resource by its absolute location
+public type ReferenceSearchParameter record {
+    *FHIRTypedSearchParameter;
+    readonly & string? resourceType;
+    readonly & string? id;
+    readonly & string? url;
+};
+
+# Composite type search parameter information record.
+# 
+# + value - Composite search parameter value
+public type CompositeSearchParameter record {
+    *FHIRTypedSearchParameter;
+    readonly string value;
+    // TODO : Further imporvement needed : https://github.com/wso2-enterprise/open-healthcare/issues/924
```

Comment:
References to internal issues are still there. But it shouldn't be an issue. Any concerns??